src/kp/kp_client.py

```diff
@@ -1,17 +1,18 @@
+import time
 import logging
 import requests
-import json
 from typing import Dict, Tuple
 
 from requests.auth import HTTPBasicAuth
 from base_api_client import BaseAPIClient
-from utils.config import KP_URL, BROWSERLESS_CLIENT_ID, BROWSERLESS_CLIENT_SECRET
+from utils.config import KP_URL, BROWSERLESS_URL, BROWSERLESS_CLIENT_ID, BROWSERLESS_CLIENT_SECRET
 
 logger = logging.getLogger(__name__)
 
 
 class KPAPIClient(BaseAPIClient):
     _client_cache: Dict[Tuple[str, str], 'KPAPIClient'] = {}
+    isFetchingToken = False
```

Comment:
Should be in init and it should be `self.isFetchingToken = False`

Reply:
Will fix.

Reply:
fix before merging

Reply:
Fixed